src/main/java/com/softserve/academy/spaced/repetition/service/CardService.java

```diff
@@ -50,7 +50,16 @@ public void updateCard(Long id, Card card) {
         return cardsQueue;
     }
 
-    public void deleteCard(Long id) {
-        cardRepository.delete(id);
+    public void deleteCard(Long deck_id, Long card_id) {
+        Deck deck = deckRepository.getDeckById(deck_id);
+        List<Card> cards = deck.getCards();
+        for(Card card: cards) {
```

Comment:
Why not getting the card by id directly?

Card cardToDelete = cardRepository.getCardById(...)

It wrong to iterate over all the cards to find one.